Source/WebCore/rendering/MarkedText.h

```diff
@@ -68,24 +68,59 @@ struct MarkedText {
         Frontmost
     };
 
-    unsigned startOffset;
-    unsigned endOffset;
-    Type type;
+    MarkedText(unsigned startOffset, unsigned endOffset, Type type, const RenderedDocumentMarker* marker = nullptr, AtomString highlightName = { }, int priority = 0) : startOffset(startOffset), endOffset(endOffset), type(type), marker(marker), highlightName(highlightName), priority(priority) { };
+    MarkedText(WTF::HashTableDeletedValueType) : m_isHashTableDeletedValue(true) { };
+    MarkedText() { };
+
+    unsigned startOffset { 0 };
+    unsigned endOffset { 0 };
+    Type type { Type::Unmarked };
     const RenderedDocumentMarker* marker { nullptr };
     AtomString highlightName { };
-
+    int priority { 0 };
     bool isEmpty() const { return endOffset <= startOffset; }
+    bool isHashTableDeletedValue() const { return m_isHashTableDeletedValue; }
     bool operator==(const MarkedText& other) const
     {
-        return startOffset == other.startOffset && endOffset == other.endOffset && type == other.type && marker == other.marker && highlightName == other.highlightName;
+        return startOffset == other.startOffset && endOffset == other.endOffset && type == other.type && marker == other.marker && priority == other.priority
+        && ((highlightName == other.highlightName) || (highlightName.isNull() && other.highlightName.isNull()));
```

Comment:
@cdumez Is the equality check enough here? or do we also need the or with both null strings? Also makes me wonder if we can use the default C++ equality operator here.

Reply:
Tim is right here, you don't need these null checks. `string1 == string2` will return true if both strings are null.